docs/tutorials/13_quantum_bayesian_inference.ipynb

```diff
@@ -0,0 +1,747 @@
+    "# Quantum Bayesian Inference\n",
+    "\n",
+    "## Overview\n",
+    "This notebook demonstrates a quantum Bayesian inference (QBI) implementations provided in `qiskit-machine-learning`, and how it can be integrated into basic quantum machine learning (QML) workflows.\n",
+    "\n",
+    "The tutorial is structured as follows:\n",
+    "\n",
+    "1. [Introduction](#1.-introduction)\n",
+    "2. [How to Instantiate QBI](#2.-how-to-instantiate-qbi)\n",
+    "3. [How to Run Rejection Sampling](#3.-how-to-run-rejection-sampling)\n",
+    "4. [How to Run an Inference](#4.-how-to-run-an-inference)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "494f210f33019c5b",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## 1. Introduction"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### 1.1. Quantum vs. Classical Bayesian Inference\n",
+    "\n",
+    "Bayesian networks, or belief networks, are graphical models that illustrate probabilistic relationships between variables using nodes (representing variables) and edges (indicating conditional dependencies) in a directed acyclic graph. Each node is associated with conditional probability tables (CPTs) that detail the influence of parent nodes on their children. \n",
+    "\n",
+    "In these networks, Bayesian inference is key for updating probabilities. It employs Bayes' theorem to revise the likelihood of hypotheses based on new data, considering the network's variable interdependencies. For instance, in a network assessing diseases based on symptoms, observing new symptoms allows for recalculating disease probabilities. This recalibration combines the disease's prior probability with the observed symptom likelihood, leading to an updated, more precise disease probability. Thus, Bayesian inference is a dynamic process of adjusting our understanding of one variable in light of new information about others, facilitating informed, evidence-based decisions.\n",
+    "\n",
+    "Exact inference on Bayesian networks is \\#P-hard. That is why usually approximate inference is used to sample from the distribution on query variables given evidence variables. QBI efficiently utilizes the structure of Bayesian networks represented by a quantum circuit that represent the probability distributions. By employing a quantum version of rejection sampling and leveraging amplitude amplification, quantum computation achieves a significant speedup, making it possible to obtain samples much faster. \n",
+    "\n",
+    "This tutorial will guide you through the process of using the QBayesian class to perform such inference tasks. This inference algorithm implements the algorithm from the paper \"Quantum inference on Bayesian networks\" by Low, Guang Hao et al. This leads to a speedup per sample from $O(nmP(e)^{-1})$ to $O(n2^{m}P(e)^{-\\frac{1}{2}})$, where n is the number of nodes in the Bayesian network with at most m parents per node and e the evidence.\n",
```

Comment:
Can you please add the reference paper here too?

Reply:
I will do so, anything else?

Reply:
I think it is ready after adressing:
> As a note, to avoid the issue we just had with another PR from last year #728 where I needed to update the copyright dates following it getting merged - see #737 I would suggest updating the copyrights in the set of files changed/added here to include 2024 (if they already have 2 dates then the last one change from 2023 to 2024 - if just has 2023 then it it should become 2023, 2024).

Reply:
Ok I added it. Please take a look if everything is good.